BLSQ/iaso
# IA-2049 fix Bug with special char in enketo edit

The problem was that BeautifulSoup didn't parse the xml properly when it contained a Ï (i uppercase with trema)  and so couldn't find the instanceId as it only parsed half of the file.

I fixed it by converted the code to parse  using lxml.etree instead of BeautifulSoup


## Changes
- [ ] Use lxml instead of beautiful soup to get the id
- [ ] Took the opportunity to simplify the logic so it doesn't parse and rewrite the XML three times.
- [ ] Added regressions test


Related JIRA tickets : IA-2049

## Self proofreading checklist

- [NA] Are my typescript files well typed
- [NA] New translations have been added or updated if new strings have been introduced in the frontend
- [x] Are there enough tests. New test yay
- [NA] Documentation has been included (for new feature)

## How to test
- Make a submission using the bad char (see fixture file included), check that you can edit it via Enketo.
- check that the other part of the Enketo feature are still working properly, try editing different submissions.


## Notes

There are other part of the code that use beautiful soup that we could also convert in the future but that outside the scope of this pr
